File: Backend/Mono/Nonagon.Modular.DynamicForm/FieldType.cs
namespace Nonagon.Modular.DynamicForm
{
	/// <summary>
	/// Field type to define the field.
	/// </summary>
	public enum FieldType
	{
		/// <summary>
		/// The free text input box.
		/// </summary>
		FreeText,

		/// <summary>
		/// The multiline text input box.
		/// </summary>
		MultilineText,

		/// <summary>
		/// The email only input box.
		/// </summary>
		Email,

		/// <summary>
		/// The phone number only input box.
		/// </summary>
		PhoneNumber,

		/// <summary>
		/// The decimal number input box.
		/// </summary>
		Decimal,

		/// <summary>
		/// The number only input box.
		/// </summary>
		Digits,

		/// <summary>
		/// The url input box.
		/// </summary>
		Url,

		/// <summary>
		/// The choices (select only one item).
		/// </summary>
		Choices,

		/// <summary>
		/// The checkbox which only one item can be checked.
		/// </summary>
		SingleCheck,

		/// <summary>
		/// The checkbox which multiple items can be checked.
		/// </summary>
		MultiChecks,

		/// <summary>
		/// The date picker.
		/// </summary>
		DatePicker,

		/// <summary>
		/// The time picker.
		/// </summary>
		TimePicker,

		/// <summary>
		/// The date and time picker.
		/// </summary>
		DateTimePicker,

		/// <summary>
		/// The picture field.
		/// </summary>
		Picture
	}
}

File: Frontend/GWT/gwt-modular-dynamicform/src/com/nonagon/modular/dynamicform/client/FieldType.java
package com.nonagon.modular.dynamicform.client;

public enum FieldType {

	FreeText,
	MultilineText,
	Email,
	PhoneNumber,
	Decimal,
	Digits,
	Url,
	Choices,
	SingleCheck,
	MultiChecks,
	DatePicker,
	TimePicker,
	DateTimePicker;
	
	public static FieldType fromString(String value) {

		if(value == null)
			return FreeText;
					
		return Enum.valueOf(FieldType.class, value);
	}
	
	public static String fromEnum(FieldType fieldType) {
		
		return fieldType.toString();
	}
}
